### PR TITLE
New API for remaining Immovable programs

### PR DESCRIPTION
**Pull request by gunchleoc**
* `seed=<immovable_name> proximity:<percent>`
* Fix datatypes for probability ranges
* Update documentation

Re #4047

### CODE REVIEW

File: src/logic/map_objects/immovable_program.cc [hunk 1]
```diff
@@ -396,49 +397,73 @@ void ImmovableProgram::ActRemove::execute(Game& game, Immovable& immovable) cons
...
-random location, and it will only seed there if the terrain is well suited. This command may be used
-only for immovables with a terrain affinity.
+     main = {
+         "animate=idle duration:20s",
+         "remove=chance:11.72%",
+         -- Select a location with a chance of 19.53% in the base radius,
+         -- then expand the radius and try again with a chance of 19.53%.
+         -- Repeat until a location has been selected, then plant an
+         -- 'umbrella_red_wasteland_sapling' if the terrain affinity check
+         -- for this immovable succeeds.
```
Comment (hessenfarmer) [inline src/logic/map_objects/immovable_program.cc]: suggestion: for this combination of location and immovable succeeds

File: src/logic/map_objects/immovable_program.cc [hunk 2]
```diff
@@ -396,49 +397,73 @@ void ImmovableProgram::ActRemove::execute(Game& game, Immovable& immovable) cons
...
+         "seed=umbrella_red_wasteland_sapling proximity:19.53%",
+         "animate=idle duration:20s",
+         "remove=chance:7.81%",
+         "grow=umbrella_red_wasteland_old",
```
Comment (hessenfarmer) [inline src/logic/map_objects/immovable_program.cc:431]: the lines after the seed command seem to be a copy paste error, as they do not result in a meaningful immovableprogram

**Review by hessenfarmer — APPROVED** [review]
Code LGTM.
2 small nits in the doc part.
There is a typo in the PR description.

File: src/logic/map_objects/immovable_program.cc (hunk 2 above)
Comment (gunchleoc) [reply src/logic/map_objects/immovable_program.cc:431]: This is 100% what's in the actual program. There's a 92.19% chance that the remove program fails, in which case the tree might grow.

File: src/logic/map_objects/immovable_program.cc (hunk 1 above)
Comment (gunchleoc) [reply src/logic/map_objects/immovable_program.cc]: Thanks! Changed to "for this immovable succeeds at the selected location."

**Pull request merged**